posts/2017-06-02-devops-serverless-variables.md

```diff
@@ -0,0 +1,155 @@
+---
+title: "DevOps Use Cases With Serverless Variables"
+description: Serverless Framework Engineer Eslam Hefnawy explores creative use cases for using Serverless Variables to optimize and automate operations.
+date: 2017-05-01
+layout: Post
+thumbnail:https://avatars2.githubusercontent.com/u/2312463?v=3&s=60
+authors:
+  - eahefnawy
+---
+
+
+The Serverless Framework has a powerful built-in variable system that helps secure your sensitive data and cab be used to keep even the most complex configuration simple and manageable. We pretty comprehensive coverage of all of the features of the variable system in [our documentation](https://serverless.com/framework/docs/providers/aws/guide/variables/#variables), but when combined together in creative ways these features enable some extremely powerful workflows and real world use cases. 
+
+In this article we’ll explore some ideas on how you can use these features to better automate your serverless operations.
+
+## Dyamic Configuration
+You can use the variable system to dynamically generate configuration data for your config files with the help of JS scripts, which is normally not supported by YAML/JSON. Some examples include generating random numbers, dates, doing mathematical computation, or fetching remote data. Let’s look at two of these examples..
+
+
+### Generating Random ID
+Let’s say you want to create a bucket with a unique id attached to the name. It wouldn’t be possible to generate that random id inside of serverless.yml without the magic of programming. You can utilize the variable system for this use case like this:
+
+```yml
+# serverless.yml
+
+Resources:
+  Resources:
+    myBucket:
+      Type: AWS::S3::Bucket
+      Properties:
+        BucketName: my-bucket-with-id-${file(vars.js):bucketId}
+```
+
+And in vars.js
+```js
+module.exports.bucketId = (new Date()).getTime().toString();
+```
+
+
+
+
+### Fetching Remote Data
+You can go even further with JS scripts by working with promises and async operations, which is supported out of the box by the framework. That means you can fetch data needed for your configuration from any remote source you want.
```

Comment:
`by the framework.` -- Do we capitalize "Framework" when talking about the Serverless Framework?